src/auto-evo/AutoEvoExploringTool.cs

```diff
@@ -418,6 +422,29 @@ public void AskExit()
         exitConfirmationDialog.PopupCenteredShrink();
     }
 
+    public Species? GetActiveSpeciesData(uint speciesId)
+    {
+        var gameWorld = world.GameProperties.GameWorld;
+
+        for (int i = generationDisplayed; i >= 0; --i)
+        {
+            gameWorld.GenerationHistory[i].AllSpeciesData
+                .TryGetValue(speciesId, out var speciesRecord);
+
+            if (speciesRecord == null)
+                return null;
```

Comment:
Now that I'm re-reading this I have one more question. What's this code for? Doesn't this cause an early return that would potentially miss data? Shouldn't this just be a case of continuing the search if there is no species record in the current generation for the wanted species?

Reply:
I did some thorough investigation on this, and found out that this indeed causes missing data in some edge-cases. More exactly, if a species that is extinct in current generation is requested, it is null. Technically, the former is impossible to do in the current Auto-Evo explorer, but still may cause problems in future. Anyway, the fix is as simple as switching `return` to `continue`.